From d0f3b8a60df4b9564b53a10fd8091e2fd72b4323 Mon Sep 17 00:00:00 2001
From: Prabhu Kuttiyam <pkuttiyam@ti.com>
Date: Fri, 7 Dec 2012 09:35:10 -0500
Subject: [PATCH 078/257] hwqueue: modify test code to match API change

This commit modifies the hwqueue_test code to match the new API syntax.

Signed-off-by: Prabhu Kuttiyam <pkuttiyam@ti.com>
(cherry picked from commit 7404a7200ffd2bb6d2d53cbc67b5d30267cc0f8d)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/hwqueue/hwqueue_test.c | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/drivers/hwqueue/hwqueue_test.c b/drivers/hwqueue/hwqueue_test.c
index 4f3aa65..06a4cb0 100644
--- a/drivers/hwqueue/hwqueue_test.c
+++ b/drivers/hwqueue/hwqueue_test.c
@@ -57,7 +57,7 @@ static void hwqueue_test_notify_irq(void *arg)
 #else
 	unsigned desc_size;
 	dma_addr_t desc;
-	desc = hwqueue_pop(ctx->qirqrd, &desc_size, NULL);
+	desc = hwqueue_pop(ctx->qirqrd, &desc_size, NULL, 0);
 	if (!desc)
 		pr_err("test3: failed to pop desc from read queue\n");
 
@@ -144,7 +144,7 @@ static int __init hwqueue_test(void)
 	}
 	ctx.notify_added = true;
 
-	desc = hwqueue_pop(qpool, &desc_size, NULL);
+	desc = hwqueue_pop(qpool, &desc_size, NULL, 0);
 	if (!desc) {
 		ret = -ENOENT;
 		pr_err("failed to pop pool desc\n");
@@ -156,14 +156,14 @@ static int __init hwqueue_test(void)
 
 	/* Test 1: Simple push and pop */
 	pr_err("test1: simple pop test:\n");
-	ret = hwqueue_push(qgenwr, desc, desc_size);
+	ret = hwqueue_push(qgenwr, desc, desc_size, 0);
 	if (ret < 0) {
 		pr_err("test1: failed to push desc, errno=%d\n", ret);
 		goto fail;
 	}
 	pr_info("test1: pushed desc %08x, size %d\n", desc, desc_size);
 
-	desc = hwqueue_pop(qgenrd, &desc_size, NULL);
+	desc = hwqueue_pop(qgenrd, &desc_size, NULL, 0);
 	if (!desc) {
 		ret = -ENOENT;
 		pr_err("test1: failed to pop desc from read queue\n");
@@ -178,7 +178,7 @@ static int __init hwqueue_test(void)
 
 	/* Test 2: Notified push and pop */
 	pr_err("test2: poll notified pop test:\n");
-	ret = hwqueue_push(qgenwr, desc, desc_size);
+	ret = hwqueue_push(qgenwr, desc, desc_size, 0);
 	if (ret < 0) {
 		pr_err("failed to push desc, errno=%d\n", ret);
 		goto fail;
@@ -195,7 +195,7 @@ static int __init hwqueue_test(void)
 		pr_err("test2: timed out waiting for hwqueue notification\n");
 	ctx.notified = false;
 
-	desc = hwqueue_pop(qgenrd, &desc_size, NULL);
+	desc = hwqueue_pop(qgenrd, &desc_size, NULL, 0);
 	if (!desc) {
 		ret = -ENOENT;
 		pr_err("test2: failed to pop desc from read queue\n");
@@ -211,7 +211,7 @@ static int __init hwqueue_test(void)
 	/* Test 3: IRQ push, notify and pop */
 	pr_err("test3: irq notified pop test:\n");
 	pr_info("test3: pushing desc %08x, size %d\n", desc, desc_size);
-	ret = hwqueue_push(qirqwr, desc, desc_size);
+	ret = hwqueue_push(qirqwr, desc, desc_size, 0);
 	if (ret < 0) {
 		pr_err("failed to push desc, errno=%d\n", ret);
 		goto fail;
@@ -228,7 +228,7 @@ static int __init hwqueue_test(void)
 	ctx.notified_irq = false;
 
 #ifdef irqpop_test1
-	desc = hwqueue_pop(qirqrd, &desc_size, NULL);
+	desc = hwqueue_pop(qirqrd, &desc_size, NULL, 0);
 	if (!desc) {
 		ret = -ENOENT;
 		pr_err("test3: failed to pop desc from read queue\n");
@@ -244,7 +244,7 @@ static int __init hwqueue_test(void)
 
 fail:
 	if (ctx.desc)
-		hwqueue_push(ctx.qpool, ctx.desc, ctx.desc_size);
+		hwqueue_push(ctx.qpool, ctx.desc, ctx.desc_size, 0);
 	if (ctx.notify_added)
 		hwqueue_set_notifier(ctx.qgenrd, NULL, NULL);
 	if (ctx.notify_irq_added)
-- 
2.7.4

